### ISSUE TITLE
Training Matcha for other languages: Fine tunning HiFiGAN to specific speakers

### ISSUE THREAD

**Issue by ibonsc** [shivammehta25/Matcha-TTS#124]
### Discussed in https://github.com/shivammehta25/Matcha-TTS/discussions/120

<div type='discussions-op-text'>

<sup>Originally posted by **ibonsc** November 28, 2024</sup>
Hello
I'm training Matcha for different languages and I think it would be good to fine tune the universal HiFiGAN vocoder to these new languages. Have you done such a thing? 
To do that it is necessary to get the ground truth-aligned spectrograms of the training material, that will be used as input for the HiFiGAN finetuning. Is there any way to do that? 
Thank you very much for your help.</div>

**Comment by shivammehta25** [comment]
Yeah! Ofcourse, what you can do it extract alignments from a trained model, using the first part of: https://github.com/shivammehta25/Matcha-TTS/wiki/Extracting-phoneme-alignments-and-improving-GPU-utilisation

Then, you would have to hack the inference code and use these alignments instead. One way I can suggest is to use the batched dataset in the CLI and also load the transcripts and durations:

https://github.com/shivammehta25/Matcha-TTS/blob/108906c603fad5055f2649b3fd71d2bbdf222eac/matcha/cli.py#L292-L300


Then use these saved durations instead of `w`
https://github.com/shivammehta25/Matcha-TTS/blob/108906c603fad5055f2649b3fd71d2bbdf222eac/matcha/models/matcha_tts.py#L121

And it should be fine. Hope this helps, let me know if you have any more doubts.
